### PR TITLE
Added new script for running oncoanalyser through orcabus

### PR DESCRIPTION
**Pull request by alexiswl**
* Use a manifest json as input

* Assumes nextflow roles have permissions to read/write to cache buckets

* Supports both DNA only (from bam), RNA only (from fastq) and DNA + RNA (from post-oncoanalyser) workflows

* Params now stored in params.json

* Use jq to build param args and samplesheet csv

### CODE REVIEW

**Comment by scwatts** [comment]
Thanks so much, and truly next level use of jq!

Have you tested this in dev? If not happy to give it a go

**Comment by scwatts** [comment]
Were there any further changes needed to switch from subject ID to library ID?

**Comment by alexiswl** [comment]
> Have you tested this in dev? If not happy to give it a go

No not yet, just getting the structure right first, we will need to also manually give the oncoanalyser roles permissions to access the data-account buckets as well before we can start testing

**Comment by alexiswl** [comment]
> Were there any further changes needed to switch from subject ID to library ID?

I don't think so, for now we can just continue to use the SBJ nfrom the linked individual object, this is set via the orchestration rather than the nextflow stack too so we can change this down the track without needing to make changes here

**Comment by scwatts** [comment]
Great, that makes it easy though I will probably change variable names from subject to library or similar when that time comes

For testing is there anything stopping us from using the current S3 bucket to avoid permissions issues? e.g. manually stage inputs into our temp dev bucket then provide an appropriate manifest to the run-v2.sh script. I'd love to test this early in the dev process if possible

File: application/pipeline-stacks/oncoanalyser/assets/run-v2.sh [hunk 1]
```diff
@@ -0,0 +1,965 @@
...
+  '
+  local manifest_json="${1}"
+
+  jq --raw-output \
+    '
+      .inputs as $inputs |
+      .$inputs.tumor_rna_fastq_uri_list |
+      map(
+        {
+          "group_id": "\($inputs.tumor_rna_sample_id)",
```
Comment (alexiswl) [inline application/pipeline-stacks/oncoanalyser/assets/run-v2.sh]: Should be .inputs not $inputs

File: application/pipeline-stacks/oncoanalyser/assets/run-v2.sh [hunk 2]
```diff
@@ -0,0 +1,965 @@
...
+          "filetype": "fastq",
+          "info": "library_id:\(.rglb);lane\(.lane)",
+          "filepath": "\(.read1File);\(.read2File)"
+        }
+      )
+    ' <<< "${manifest_json}" | \
+    jq_to_csv > "${SAMPLESHEET_CSV_PATH}"
+}
+
+generate_wgts_dna_rna_to_samplesheet(){
```
Comment (alexiswl) [inline application/pipeline-stacks/oncoanalyser/assets/run-v2.sh:435]: Need to replace these with sash inputs

File: application/pipeline-stacks/oncoanalyser/assets/run-v2.sh [hunk 3]
```diff
@@ -0,0 +1,965 @@
...
+     "normal_dna_bam_uri": "<normal_bam_uri>",  // Required if analysis_type is set to DNA
+     "tumor_rna_sample_id": "<rna_sample_id>",  // Required if analysis_type is set to RNA
+     "tumor_rna_fastq_uri_list": [ <Array of wts fastq list rows> ]  // Required if analysis_type is set to RNA
+     "dna_oncoanalyser_analysis_uri": "<oncoanalyser_dir>"  // Required if analysis_type is set to DNA/RNA
+     "rna_oncoanalyser_analysis_uri": "<oncoanalyser_dir>"  // Required if analysis_type is set to DNA/RNA
+   },
+   "engine_parameters": {
+     "portal_run_id": "<portal_run_id>",  // Always required
+     "output_results_dir": "<output_results_dir>",  // Always required
+     "output_staging_dir": "<output_staging_dir>",  // Always required
```
Comment (alexiswl) [inline application/pipeline-stacks/oncoanalyser/assets/run-v2.sh]: Staging dir no longer required

File: application/pipeline-stacks/oncoanalyser/assets/run-v2.sh [hunk 4]
```diff
@@ -0,0 +1,965 @@
...
+{
+   "inputs": {
+     "mode": "wgts | targeted"
+     "analysis_type": "DNA | RNA | DNA/RNA"
+     "subject_id": "<subject_id>", // Required
+     "tumor_dna_sample_id": "<tumor_sample_id>",  // Required if analysis_type is set to DNA or DNA/RNA
+     "normal_dna_sample_id": "<normal_sample_id>",  // Required if analysis_type is set to DNA or DNA/RNA
+     "tumor_dna_bam_uri": "<tumor_bam_uri>",  // Required if analysis_type is set to DNA
+     "normal_dna_bam_uri": "<normal_bam_uri>",  // Required if analysis_type is set to DNA
+     "tumor_rna_sample_id": "<rna_sample_id>",  // Required if analysis_type is set to RNA
```
Comment (alexiswl) [inline application/pipeline-stacks/oncoanalyser/assets/run-v2.sh]: Also required when analysis_type is set to DNA/RNA

**Comment by victorskl** [comment]
(subscribed)

**Comment by victorskl** [comment]
@scwatts We had a discussion/update yesterday in devops. Just want to loop you in.

This PR intertwined with [OrcaBus O3 release](https://umccr.slack.com/archives/C03ABJTSN7J/p1726620818073309?thread_ts=1724307099.925769&cid=C03ABJTSN7J).

I understood that changes contain in this PR, added extension to OrcaBus automation (and doest not affect to existing ops with Portal automation).

So, we probably need to deploy this to prod after testing has performed (tbc in `#arteria-dev` and `#alerts-dev` for events). 

FYI

File: application/pipeline-stacks/oncoanalyser/Dockerfile [hunk 5]
```diff
@@ -40,11 +41,16 @@ RUN \
   apt update -y -q && \
   apt install -y -q \
     curl \
-    jq="${JQ_VERSION}*" \
```
Comment (scwatts) [inline application/pipeline-stacks/oncoanalyser/Dockerfile:43]: What is the reason for switching to a manual install for jq?

**Review by scwatts — APPROVED** [review]
Thanks for all the work on this and the various fixes / improvements

**Review by victorskl — APPROVED** [review]
LGTM as well.

@alexiswl Pls merge when you ready.

I reckon, the CodePipeline should deploy (again) to dev.

File: application/pipeline-stacks/oncoanalyser/Dockerfile (hunk 5 above)
Comment (alexiswl) [reply application/pipeline-stacks/oncoanalyser/Dockerfile:43]: Debian 11 installs jq v1.6 by default but we need v1.7 for the following:

```
      .inputs.mode as $mode |
      if any($valid_modes_array[] == $mode) then
        $mode
      else
        null
      end
```

**Comment by alexiswl** [comment]
> I reckon, the CodePipeline should deploy (again) to dev.

I don't think the codepipeline will deploy to dev again, 

It deploys specifically to staging and prod

https://github.com/umccr/nextflow-stack/blob/main/deployment/codepipeline-stack.ts#L95-L119

File: application/pipeline-stacks/oncoanalyser/Dockerfile (hunk 5 above)
Comment (scwatts) [reply application/pipeline-stacks/oncoanalyser/Dockerfile:43]: Got it, I'll move the jq install to Conda outside this PR after merging so we don't have to rely on manual installation

**Pull request merged**